model/project_ref.go

```diff
@@ -405,28 +406,26 @@ func FindMergedProjectRef(identifier string) (*ProjectRef, error) {
 }
 
 func mergeBranchAndRepoSettings(pRef *ProjectRef, repoRef *RepoRef) *ProjectRef {
-	// if a setting is disabled overall, then disable the branch
-	pRef.Owner = repoRef.Owner
-	pRef.Repo = repoRef.Repo
-	if !repoRef.Enabled {
-		pRef.Enabled = repoRef.Enabled
-	}
-	if repoRef.PatchingDisabled {
-		pRef.PatchingDisabled = repoRef.PatchingDisabled
-	}
-	if repoRef.RepotrackerDisabled {
-		pRef.RepotrackerDisabled = repoRef.RepotrackerDisabled
-	}
-	if pRef.DispatchingDisabled {
-		pRef.DispatchingDisabled = repoRef.DispatchingDisabled
-	}
-	// if the following fields aren't configured, default to the repo configuration
-	if pRef.RemotePath == "" {
-		pRef.RemotePath = repoRef.RemotePath
+	// if the setting is not defined in the project, default to the repo settings.
+	// For booleans, we default to the repo setting if the boolean is false, except for enabled.
+	reflectedBranch := reflect.ValueOf(pRef)
+	reflectedRepo := reflect.ValueOf(repoRef)
+
+	isProjectDisabled := !pRef.Enabled
+	for i := 0; i < reflectedBranch.Elem().NumField(); i++ {
+		branchField := reflectedBranch.Elem().FieldByIndex([]int{i})
+
+		if branchField.IsZero() {
+			reflectedField := reflectedRepo.Elem().FieldByIndex([]int{0, i})
+			reflectedBranch.Elem().Field(i).Set(reflectedField)
+		}
 	}
-	if pRef.SpawnHostScriptPath == "" {
-		pRef.SpawnHostScriptPath = repoRef.SpawnHostScriptPath
+
+	// if the branch was explicitly disabled, override any repo setting
+	if isProjectDisabled {
```

Comment:
I wonder if it makes more sense to have this in the loop since another way to think of it is "the disabled flag never inherits from the repo." It might be a moot point if you switch to a bool pointer though

Reply:
I would argue it's not true that the disabled flag never inherits from the repo, because we may still be loading what _would_ be true, but yeah probably moot.